Sources/Components/IconCollection/IconCollectionView.swift

```diff
@@ -51,6 +51,35 @@ public final class IconCollectionView: UIView {
         addSubview(collectionView)
         collectionView.fillInSuperview()
     }
+
+    // MARK: - Overrides
+
+    // This override exists because of how we calculate view sizes in our objectPage.
+    // The objectPage needs to know the size of this view before it's added to the view hierarchy, aka. before
+    // the collectionView itself knows it's own contentSize, so we need to calculate the total height of the view manually.
+    //
+    // All we're given to answer this question is the width attribute in `targetSize`.
+    //
+    // This implementation may not work for any place other than the objectPage, because:
+    //   - it assumes `targetSize` contains an accurate targetWidth for this view.
+    //   - it ignores any potential targetHeight.
+    //   - it ignores both horizontal and vertical fitting priority.
+    public override func systemLayoutSizeFitting(_ targetSize: CGSize, withHorizontalFittingPriority horizontalFittingPriority: UILayoutPriority, verticalFittingPriority: UILayoutPriority) -> CGSize {
+        let targetWidth = targetSize.width
+        let cellWidths = viewModels.map { cellWidth(forWidth: targetWidth, viewModel: $0) }
+
+        let cellSizes = zip(viewModels, cellWidths).map { (viewModel, width) -> CGSize in
+            let height = IconCollectionViewCell.height(for: viewModel, withWidth: width)
+            return CGSize(width: width, height: height)
+        }
+
+        guard let firstItem = cellSizes.first else { return targetSize }
+        let numberOfCellsInRow = Int(floor(targetWidth / firstItem.width))
+        let cellRows = cellSizes.chunked(into: numberOfCellsInRow)
+        let totalHeight = cellRows.compactMap { $0.max(by: { $0.height < $1.height }) }.reduce(0, { $0 + $1.height })
+
+        return CGSize(width: targetWidth, height: totalHeight + .mediumLargeSpacing)
```

Comment:
Why do we need `.mediumLargeSpacing` here?

Reply:
Just to add some spacing to the bottom. A few cases I noticed had the collectionView "too close" to the ad's description.